src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrPusher.cs

```diff
@@ -0,0 +1,168 @@
+        }  
+    }
+
+    private void ExecutePush(string remoteName, string branchName)
+    {
+        using var vmrRepo = new Repository(_vmrInfo.VmrPath);
+        vmrRepo.Config.Add("user.name", Constants.DarcBotName);
+        vmrRepo.Config.Add("user.email", Constants.DarcBotEmail);
+
+        var remote = vmrRepo.Network.Remotes[remoteName];
+        if (remote == null)
+        {
+            throw new Exception($"No remote named {remoteName} found in VMR repo.");
+        }
+
+        var branch = vmrRepo.Branches[branchName];
+        if (branch == null)
+        {
+            _logger.LogDebug($"No branch {branchName} found in VMR repo. Creating a new branch {branchName}");
+            branch = vmrRepo.CreateBranch(branchName);
+        }
+
+        var pushOptions = new PushOptions
+        {
+            CredentialsProvider = (url, user, cred) =>
+                new UsernamePasswordCredentials
+                {
+                    Username = _vmrRemoteConfiguration.GitHubToken,
+                    Password = string.Empty
+                }
+        };
+
+        vmrRepo.Network.Push(remote, branch.CanonicalName, pushOptions);
+    }
+
+    private async Task<bool> CheckCommitAvailability(string gitHubApiPat, CancellationToken cancellationToken)
+    {
+        var commitsSearchArguments = _sourceManifest
+            .Repositories
+            .Select(r => (GitRepoUrlParser.GetRepoNameAndOwner(r.RemoteUri), r.CommitSha))
+            .Select(r => new CommitSearchArguments(r.Item1.RepoName, r.Item1.Org, r.CommitSha));
+
+        var commits = await GetGitHubCommits(commitsSearchArguments, gitHubApiPat, cancellationToken);
+        
+        if (commits == null || commits.Data == null)
+        {
+            _logger.LogError("The Graphql query is not in the correct format.");
+            return false;
+        }
+
+        foreach (var pair in commits.Data)
+        {
+            if (pair.Value == null)
+            {
+                var commitArgs = commitsSearchArguments.First(r => GetGraphQlIdentifier(r.RepoName) == pair.Key);
+                _logger.LogError($"Repository {Constants.GitHubUrlPrefix}{commitArgs.RepoOwner}/{commitArgs.RepoName} was not found!");
+                return false;
+            }
+
+            if (pair.Value.Object == null)
+            {
+                var commitArgs = commitsSearchArguments.First(r => GetGraphQlIdentifier(r.RepoName) == pair.Key);
+                _logger.LogError($"Commit {commitArgs.Sha} was not found in {Constants.GitHubUrlPrefix}{commitArgs.RepoOwner}/{commitArgs.RepoName}!");
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private async Task<CommitsQueryResult?> GetGitHubCommits(
+        IEnumerable<CommitSearchArguments> commits,
+        string gitHubApiPat,
+        CancellationToken cancellationToken)
+    {
+        var queries = commits
```

Comment:
Please add a comment of what we're querying and what we expect back

Reply:
ok

Reply:
I'll write a summary